File: 02-tech/03-software-technologies/04-javascript/03-blog-basic-functionality/blog-basic-design/models/comment.js
// jshint esversion:6
const Sequelize = require('sequelize');

/**
 *
 * @param {Sequelize.Sequelize} sequelize
 * @param {*} dataTypes
 * @returns { Sequelize.Model }
 */
function createComment(sequelize, dataTypes)
{
    const Comment = sequelize.define('Comment',
    {
        content:
        {
            type: Sequelize.STRING,
            allowNull: false,
            // @ts-ignore
            required: true
        },
        date:
        {
            type: Sequelize.DATE,
            allowNull: false,
            required: true,
            defaultValue: Sequelize.NOW
        }
    },
    {
        timestamps: false
    });

    Comment.associate = (models) =>
    {
        Comment.belongsTo(models.Article,
        {
            foreignKey: 'articleId',
            targetKey: 'id'
        });

        Comment.belongsTo(models.User,
        {
            foreignKey: 'authorId',
            targetKey: 'id'
        });
    };

    return Comment;
}

module.exports = createComment;

File: 02-tech/03-software-technologies/07-ccsharp/02-blog/Blog/Blog/Models/Comment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Blog.Models
{
    public class Comment
    {
        public Comment() { }

        public Comment(string content, string authorId, int articleId)
        {
            this.Content = content;
            this.AuthorId = authorId;
            this.ArticleId = articleId;
            this.DateAdded = DateTime.Now;
        }

        [Key]
        public int Id { get; set; }

        [Required]
        public string Content { get; set; }

        [ForeignKey("Article")]
        public int ArticleId { get; set; }

        public Article Article { get; set; }

        [ForeignKey("Author")]
        public string AuthorId { get; set; }

        public virtual ApplicationUser Author { get; set; }

        public DateTime DateAdded { get; set; }

        public bool IsAuthor(string name)
        {
            return this.Author.UserName == name;
        }
    }
}
